### PR TITLE
Added checksum validation

### LINKED ISSUE #109: Images are never redownloaded
## Description

If a bastion node uses an earlier version of cluster-manager to provision nodes with an earlier version of OCP/RHCOS, then the cluster-manager is updated via `farosctl update` and a redeploy is attempted, the nodes (and bootstrap) will still PXE boot the older version of RHCOS. The download of those images [here](https://github.com/project-faros/cluster-manager/blob/master/app/roles/rhcos-images/tasks/main.yml#L7) only validates that the path exists before moving on, not that the destination file matches the source file. This is good because it prevents redownloading the sources every time to validate that they match the destinations, but it means that files that are in place will not be updated if they are out of date.

## Reproducing

- Install faros and cluster-manager v0.8.0
- Deploy a cluster
- Update cluster-manager to v4.6.0 with `farosctl update`
- Attempt to deploy a cluster

### Expected behavior

A new version of RHCOS is downloaded and OCP 4.6.1 is deployed to the nodes

### Observed behavior

RHCOS 4.4 is PXE booted and fails to make it out of initrd because the live rootfs image method used in RHCOS 4.6 isn't present at all in 4.4.

## Proposed solution

Parse the [sha256sum.txt](https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.6/4.6.1/sha256sum.txt) file for the target release and map the sums to the files downloaded in the [task loop](https://github.com/project-faros/cluster-manager/blob/master/app/roles/rhcos-images/tasks/main.yml#L7) to be used in the [checksum parameter](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/get_url_module.html#parameter-checksum) of get_url.

This will result in the checksums for the image files being calculated on each run of the role and compared against the checksums listed for those images on the release mirror. If the checksums match, the task will return an `ok` status. If the checksums don't match, they will be deleted and redownloaded with a `changed` status.

A pull request is incoming to support this solution.

### PR DESCRIPTION
**Pull request by jharmison-redhat**
- Downloaded sha256sum.txt from release mirror for version
- Templated a list that included matching sha256sums
- Adjusted loop to accept templated list

Fixes #109

### CODE REVIEW

**Comment by jharmison-redhat** [comment]
Quick note - I did not yet test this with a full cluster upgrade. It might be valuable to do so with 4.6.1 -> 4.6.8. I only tested with a rigged environment, not a real cluster.

**Comment by jharmison-redhat** [comment]
Update - tested on a redeploy with 4.6.8, working fine.

File: app/roles/rhcos-images/tasks/main.yml [hunk 1]
```diff
@@ -4,20 +4,33 @@
     state: directory
     mode: 0755
 
+- name: download rhcos image checksums
```
Comment (rmkraus) [inline app/roles/rhcos-images/tasks/main.yml:7]: Could we add a `changed_when: False` to this task?

File: app/roles/rhcos-images/tasks/main.yml [hunk 2]
```diff
@@ -4,20 +4,33 @@
...
+  uri:
+    url: "{{ rhcos_images_repo }}/sha256sum.txt"
+    return_content: yes
+  register: release_checksums
+
+- name: build image list
```
Comment (rmkraus) [inline app/roles/rhcos-images/tasks/main.yml:13]: Same here.

**Review by rmkraus — CHANGES_REQUESTED** [review]

File: app/roles/rhcos-images/tasks/main.yml (hunk 1 above)
Comment (jharmison-redhat) [reply app/roles/rhcos-images/tasks/main.yml:7]: Sure, agree.

File: app/roles/rhcos-images/tasks/main.yml (hunk 2 above)
Comment (jharmison-redhat) [reply app/roles/rhcos-images/tasks/main.yml:13]: I don't believe set_fact will ever report changed

Comment (rmkraus) [reply app/roles/rhcos-images/tasks/main.yml:14]: :+1

**Review by rmkraus — APPROVED** [review]

**Pull request merged**